src/link.c

```diff
@@ -501,11 +501,15 @@ static void LinkTable_invalid_reset(LinkTable *linktbl)
 
 void LinkTable_free(LinkTable *linktbl)
 {
-    for (int i = 0; i < linktbl->num; i++) {
-        FREE(linktbl->links[i]);
+    if (linktbl)
```

Comment:
The pointer to a freed linktbl might not be ``NULL``...

Reply:
My point is that if you could potentially free a LinkTable that is already freed.

Reply:
Do my changes create this potential or could that happen regardless of the changes I made? It seems that this could happen with the original code too.

Reply:
> Do my changes create this potential or could that happen regardless of the changes I made?

Your change increases this potential, because I didn't go around to the ``LinkTable``s to systematically free everything.

Reply:
By the look of it, everything should work. Your ``LinkTable_free`` improves on my previous implementation. I am going to leave this for now. I think I should really write tests for this filesystem and run them on GitHub workflow.